zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/io/CollectionKeySerializer.java

```diff
@@ -0,0 +1,72 @@
+package com.github.onsdigital.zebedee.keyring.io;
+
+import com.github.onsdigital.zebedee.keyring.CollectionKey;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.lang.reflect.Type;
+import java.util.Base64;
+
+import static java.text.MessageFormat.format;
+
+/**
+ * CollectionKeySerializer is custom {@link com.google.gson.Gson} marshall/unmarshaller for {@link CollectionKey}
+ * objects. Implements both {@link JsonSerializer} and {@link JsonDeserializer} interfaces.
+ */
+public class CollectionKeySerializer implements JsonDeserializer<CollectionKey>, JsonSerializer<CollectionKey> {
+
+    static final String SECRET_KEY_FIELD = "secret_key";
+    static final String COLLECTION_ID_FIELD = "collection_id";
+    static final String ENCRYPTION_ALGORITHM = "AES";
+    static final String FIELD_MISSING_ERR = "error deserializing CollectionKey expected {0} field but none found";
+    static final String FIELD_EMPTY_ERR = "error deserializing CollectionKey expected {0} field but was empty";
+
+    @Override
+    public CollectionKey deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        JsonObject jObj = json.getAsJsonObject();
+
+        JsonElement element = jObj.get(COLLECTION_ID_FIELD);
+        if (element == null) {
+            throw new JsonParseException(format(FIELD_MISSING_ERR, COLLECTION_ID_FIELD));
+        }
+
+        String collectionID = element.getAsString();
+        if (StringUtils.isEmpty(collectionID)) {
+            throw new JsonParseException(format(FIELD_EMPTY_ERR, COLLECTION_ID_FIELD));
+        }
+
+        element = jObj.get(SECRET_KEY_FIELD);
+        if (element == null) {
+            throw new JsonParseException(format(FIELD_MISSING_ERR, SECRET_KEY_FIELD));
+        }
+
+        String secretKeyStr = element.getAsString();
+        if (StringUtils.isEmpty(secretKeyStr)) {
+            throw new JsonParseException(format(FIELD_EMPTY_ERR, SECRET_KEY_FIELD));
+        }
+
+        byte[] keyBytes = Base64.getDecoder().decode(secretKeyStr);
+        SecretKey key = new SecretKeySpec(keyBytes, 0, keyBytes.length, ENCRYPTION_ALGORITHM);
+
```

Comment:
should zero the keyBytes array after use (have assumed it contains a raw private key).

Reply:
`keyBytes` is defined within the scope of this method so the variable will be discarded after the method returns so I think it's probably unnecessary.
That said it will technically still be on the heap until its garbage collected. I am not sure how much of a risk this is? I am happy to blitz the array if we have any concern this is a potential risk.

Reply:
Useful discussion here: https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
Will see if there are other viewpoints.

Reply:
This is a very interesting point and that link you have shared is pretty useful & insightful.

However, this presents a bit of problem - I am not sure if it is possible to read or write a `SecretKey` to a json file without creating a string object to hold it. Googling around the approach to storing a []byte in json is to base64 encode it and write it as a String (which is what I have done). However for the reasons covered in that thread it is potentially a security issue as it means the key will have been stored in a String object at some point and its therefore potentially accessible.

I don't want to say it's not possible but Im not sure how to get around it? Open to ideas & or a chat if anyone wants to discuss in detail.